plugins/texteditor.koplugin/main.lua

```diff
@@ -0,0 +1,498 @@
+                text = T(_("This file is %2:\n\n%1\n\nAre you sure you want to open it?\n\nOpening big files may take some time."),
+                    file_path, util.getFriendlySize(attr.size)),
+                ok_text = _("Yes"),
+                cancel_text = _("No"),
+                ok_callback = function()
+                    self:editFile(file_path, readonly)
+                end,
+            })
+        else
+            self:editFile(file_path, readonly)
+        end
+    else -- File does not exist
+        -- Try to create it just to check if writting to it later is possible
+        local file, err = io.open(file_path, "wb")
+        if file then
+            -- Clean it, we'll create it again on Save, and allow closing
+            -- without saving in case the user has changed his mind.
+            file:close()
+            os.remove(file_path)
+            self:editFile(file_path)
+        else
+            UIManager:show(InfoMessage:new{
+                text = T(_("This file can not be created:\n\n%1\n\nReason: %2"), file_path, err)
+            })
+            return
+        end
+    end
+end
+
+function TextEditor:readFileContent(file_path)
+    local file = io.open(file_path, "rb")
+    if not file then
+        -- We checked file existence before, so assume it's
+        -- because it's a new file
+        return ""
+    end
+    local file_content = file:read("*all")
+    file:close()
+    return file_content
+end
+
+function TextEditor:saveFileContent(file_path, content)
+    local file, err = io.open(file_path, "wb")
+    if file then
+        file:write(content)
+        file:close()
+        logger.info("TextEditor: saved file", file_path)
+        return true
+    end
+    logger.info("TextEditor: failed saving file", file_path, ":", err)
+    return false, err
+end
+
+function TextEditor:deleteFile(file_path)
+    local ok, err = os.remove(file_path)
+    if ok then
+        logger.info("TextEditor: deleted file", file_path)
+        return true
+    end
+    logger.info("TextEditor: failed deleting file", file_path, ":", err)
+    return false, err
+end
+
+function TextEditor:editFile(file_path, readonly)
+    self:addToHistory(file_path)
+    local directory, filename = util.splitFilePathName(file_path) -- luacheck: no unused
+    local filename_without_suffix, filetype = util.splitFileNameSuffix(filename) -- luacheck: no unused
+    local is_lua = filetype:lower() == "lua"
+    local input
+    input = InputDialog:new{
+        title =  filename,
+        input = self:readFileContent(file_path),
+        input_face = Font:getFace(self.font_face, self.font_size),
+        fullscreen = true,
+        condensed = true,
+        allow_newline = true,
+        cursor_at_end = false,
+        readonly = readonly,
+        add_nav_bar = true,
+        buttons = is_lua and {{
+            -- First button on first row, that will be filled with Reset|Save|Close
+            {
+                text = _("Check lua"),
```

Comment:
Why is Lua lowercase?

Reply:
I remember there was something about lua, thought it was meant to be lowercase. But no, it's just not to be made uppercase :| Correcting.